File: app/src/main/java/com/bing/example/model/db/dao/VideoDao.java
package com.bing.example.model.db.dao;


import com.bing.example.model.entity.VideoInfo;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface VideoDao {
        @Query("SELECT * FROM videos")
        List<VideoInfo> loadAllVideos();

        @Query("SELECT * FROM videos")
        LiveData<List<VideoInfo>> loadAllVideosLiveData();

        @Insert(onConflict = OnConflictStrategy.REPLACE)
        void insertAll(List<VideoInfo> details);

        @Insert(onConflict = OnConflictStrategy.REPLACE)
        void insert(VideoInfo detail);

        @Query("delete from videos where 1=1")
        void deleteAll();

        @Query("delete from videos where id=(:id)")
        void deleteVideos(int id);

        @Delete
        void deleteVideos(List<VideoInfo> videoInfos);

        @Delete
        void deleteVideo(VideoInfo videoInfo);

        @Update
        void updateVideo(VideoInfo videoInfo);
}

File: app/src/main/java/com/bing/example/model/db/dao/VideoDao.kt
package com.bing.example.model.db.dao


import com.bing.example.model.entity.VideoInfo

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update

@Dao
interface VideoDao {
        @Query("SELECT * FROM videos")
        fun loadAllVideos(): List<VideoInfo>

        @Query("SELECT * FROM videos")
        fun loadAllVideosLiveData(): LiveData<List<VideoInfo>>

        @Insert(onConflict = OnConflictStrategy.REPLACE)
        fun insertAll(details: List<VideoInfo>)

        @Insert(onConflict = OnConflictStrategy.REPLACE)
        fun insert(detail: VideoInfo)

        @Query("delete from videos where 1=1")
        fun deleteAll()

        @Query("delete from videos where id=(:id)")
        fun deleteVideos(id: Int)

        @Delete
        fun deleteVideos(videoInfos: List<VideoInfo>)

        @Delete
        fun deleteVideo(videoInfo: VideoInfo)

        @Update
        fun updateVideo(videoInfo: VideoInfo)
}
